framework/contracts/account/src/execution.rs

```diff
@@ -16,20 +16,15 @@ use crate::{
 };
 
 /// Check that sender either whitelisted or governance
-pub(crate) fn assert_whitelisted_or_owner(deps: &mut DepsMut, sender: &Addr) -> AccountResult<()> {
-    #[cfg(feature = "xion")]
-    {
-        if let Some(is_admin) = crate::state::AUTH_ADMIN.may_load(deps.storage)? {
-            // Clear auth if it was set
-            crate::state::AUTH_ADMIN.remove(deps.storage);
-            if is_admin {
-                return Ok(());
-            }
-        }
-    }
+pub(crate) fn assert_whitelisted_or_owner(
+    deps: &mut DepsMut,
+    env: &Env,
+    sender: &Addr,
+) -> AccountResult<()> {
     let whitelisted_modules = WHITELISTED_MODULES.load(deps.storage)?;
     if whitelisted_modules.0.contains(sender)
         || ownership::assert_nested_owner(deps.storage, &deps.querier, sender).is_ok()
+        || sender == env.contract.address
```

Comment:
This adds a self check so we should update the fn name to include `or_self`.

Reply:
We remove the `AUTH_ADMIN.remove` call above. Does this change the semantics for authorizated executions on XION?

Reply:
Also - we should have a test that specifically hits this case where the account calls into itself.

Reply:
We do have a test for that, in the only test that was added for this PR
Changing the name of the function
yes, the `AUTH_ADMIN.remove` call was a first idea on how to handle this variable that was not removed when we decided to place at the end of the execute function